Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/components/ToolbarController.java

```diff
@@ -0,0 +1,206 @@
+                    userOptions.setLayoutHierarchy(true);
+                    userOptions.setLayoutDagre(false);
+                    break;
+                case LAYOUT_DAGRE_TB:
+                    userOptions.setLayoutHierarchy(false);
+                    userOptions.setLayoutDagre(true);
+                    break;
+                }
+                parent.changeLayout();
+            }
+        };
+        
+        layoutHierarchy.addEventListener("onClick", layoutListener);
+        
+        layoutDagreTopBottom.addEventListener("onClick", layoutListener);
+
+        fitScreen.addEventListener("onClick", new EventListener<Event>() {
+            @Override
+            public void onEvent(Event event) throws Exception {
+                parent.fitVisualizationToWindow();
+            }
+        });
+
+        filterClear.addEventListener("onClick", e -> onClearFilter());
+        
+        filter.addEventListener("onInvoke", e -> parent.openLogFilter(e));
+        filter.addEventListener("onInvokeExt", e -> parent.openLogFilter(e));
+        filter.addEventListener(Events.ON_CLICK, e -> parent.openLogFilter(e));
+        
+        animate.addEventListener("onClick", e -> parent.openAnimation(e));
+
+        exportFilteredLog.addEventListener("onExport", e -> parent.openLogExport(e));
+        
+        exportBPMN.addEventListener("onClick", e -> parent.openBPMNExport(e));
+        
+        downloadPDF.addEventListener("onClick", new EventListener<Event>() {
+            @Override
+            public void onEvent(Event event) throws Exception {
+                parent.exportPDF();
+            }
+        });
+        
+        downloadPNG.addEventListener("onClick", new EventListener<Event>() {
+            @Override
+            public void onEvent(Event event) throws Exception {
+                parent.exportPNG();
+            }
+        });
+        
+        downloadJSON.addEventListener("onClick", new EventListener<Event>() {
+            @Override
+            public void onEvent(Event event) throws Exception {
+                parent.exportJSON();
+            }
+        });
+
+        share.addEventListener("onClick", event -> {
+            parent.openSharingWindow();
+        });
+    }
+    
+    @Override
+    public void setDisabled(boolean disabled) {
+        layoutHierarchy.setDisabled(disabled);
+        layoutDagreTopBottom.setDisabled(disabled);
+        filter.setDisabled(disabled);
+        filterClear.setDisabled(disabled || parent.getLogData().isCurrentFilterCriteriaEmpty());
+        searchText.setDisabled(disabled);
+        shortcutButton.setDisabled(disabled);
+    }
+    
+    public void setDisabledAnimation(boolean disabled) {
+        animate.setDisabled(disabled);
+    }
+    
+    public void setDisabledFilterClear(boolean disabled) {
+        filterClear.setDisabled(disabled);
+    }
+    
+    @Override
+    public void updateUI(Object data) throws Exception {
+        //
+    }

```

Comment:
This does the same thing that AbstractController does, so it's not strictly needed here.